### PR TITLE
etcd-tester: restart stresser after cleaning

### PR DESCRIPTION
**Pull request by gyuho**
```
2016-08-25 01:24:08.008214 I | etcd-tester: [round#20568 case#1] starting the stressers...
2016-08-25 01:24:08.008408 I | etcd-tester: stresser "10.240.0.2:2379" is started
2016-08-25 01:24:08.008526 I | etcd-tester: stresser "10.240.0.3:2379" is started
2016-08-25 01:24:08.008645 I | etcd-tester: stresser "10.240.0.4:2379" is started
2016-08-25 01:24:08.008676 I | etcd-tester: [round#20568 case#1] started stressers

2016-08-25 01:24:08.008701 I | etcd-tester: [round#20568 case#1] cleaning up...
```

should be

```
2016-08-25 01:24:08.008701 I | etcd-tester: [round#20568 case#1] cleaning up...

2016-08-25 01:24:08.008214 I | etcd-tester: [round#20568 case#1] starting the stressers...
2016-08-25 01:24:08.008408 I | etcd-tester: stresser "10.240.0.2:2379" is started
2016-08-25 01:24:08.008526 I | etcd-tester: stresser "10.240.0.3:2379" is started
2016-08-25 01:24:08.008645 I | etcd-tester: stresser "10.240.0.4:2379" is started
2016-08-25 01:24:08.008676 I | etcd-tester: [round#20568 case#1] started stressers
```

Otherwise, failure test data contains the writes after revision/hash mismatch, which we don't need for investigating issues.

### CODE REVIEW

**Comment by xiang90** [comment]
> restart stresser before cleaning

Why? We should NOT restart stresser before cleaning.

**Comment by gyuho** [comment]
@xiang90 Oh I said it the other way. This commit makes it restart after cleaning.

File: tools/functional-tester/etcd-tester/tester.go [hunk 1]
```diff
@@ -176,6 +176,10 @@ func (tt *tester) checkConsistency() (failed bool, err error) {
 	if !ok || err != nil {
 		plog.Printf("%s checking current revisions failed [revisions: %v]", tt.logPrefix(), revs)
 		failed = true
+		cerr := tt.cleanup()
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: can we change line148 instead?

``` go
if err == nil {
    serr = tt.start ...
    if serr != nil {
        return serr
    }
    return err
}
```

Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: we have exact same issue for compact.

Comment (gyuho) [inline tools/functional-tester/etcd-tester/tester.go]: Do you mean 

```go

func (tt *tester) checkConsistency() (failed bool, err error) {
	tt.cancelStressers()
	defer func() {
		if err == nil {
			err = tt.startStressers()
		}
	}()
```

?

We still need to run stressers even with the errors for next rounds?

Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: > We still need to run stressers even with the errors for next rounds?

I do not quite understand. We should not restart stressor if there is an error. We should always restart the stressor for a new round. The new round happens after the cleanup if there is any failure.

**Comment by xiang90** [comment]
@gyuho Please revert the cleanup change. we should do clean up at one place. we only need to stop restarting stresser on error.

**Comment by gyuho** [comment]
@xiang90 My question is 

Currently we set `err == nil` for the revision/hash mismatch (only `failed == true`). So if we do 

>  we only need to stop restarting stresser on error

when hash mismatch happens, we still clean up after restarting the stresser

Should we set error for mismatched revision or hashes?

Thanks.

**Comment by xiang90** [comment]
@gyuho see https://github.com/coreos/etcd/pull/6281

**Pull request closed without merging**